File: 스택/BOJ_10828.cpp
#include <iostream>
using namespace std;
#include <cstring>
#include <stack>

int arrayInt[100000];
int pos;
int main()
{
    ios::sync_with_stdio(0);
    cin.tie(0);
    // stack<int> S;
    // int N;
    // cin >> N;
    // while (N--)
    // {
    //     char command[10];
    //     cin >> command;
    //     if (strcmp(command, "push") == 0)
    //     {
    //         int num;
    //         cin >> num;
    //         S.push(num);
    //     }
    //     else if (strcmp(command, "top") == 0)
    //     {
    //         if (S.size() != 0)
    //             cout << S.top() << "\n";
    //         else
    //             cout << -1 << "\n";
    //     }
    //     else if (strcmp(command, "empty") == 0)
    //     {
    //         if (S.size() == 0)
    //             cout << 1 << '\n';
    //         else
    //             cout << 0 << "\n";
    //     }
    //     else if (strcmp(command, "size") == 0)
    //         cout << S.size() << "\n";
    //     else
    //     {
    //         if (S.size() != 0)
    //         {
    //             cout << S.top() << "\n";
    //             S.pop();
    //         }
    //         else
    //             cout << -1 << "\n";
    //     }
    // }

    int N;
    cin >> N;

    while (N--)
    {
        string command;
        cin >> command;
        if (command == "push")
        {
            int num;
            cin >> num;
            arrayInt[pos] = num;
            pos++;
        }
        else if (command == "top")
        {
            if (pos != 0)
                cout << arrayInt[pos - 1] << "\n";
            else
                cout << -1 << "\n";
        }
        else if (command == "empty")
        {
            if (pos == 0)
                cout << 1 << '\n';
            else
                cout << 0 << "\n";
        }
        else if (command == "size")
            cout << pos << "\n";
        else
        {
            if (pos != 0)
            {
                cout << arrayInt[pos - 1] << "\n";
                pos--;
            }
            else
                cout << -1 << "\n";
        }
    }
}

File: 스택/BOJ_10828.java
package 스택;
import java.io.*;
import java.util.*;
public class BOJ_10828 {
    static int n;
    static String order;
    static int num;
    static Stack<Integer> stack=new Stack<>();
    static StringBuilder sb=new StringBuilder();
    public static void main(String[] args)throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        n=Integer.parseInt(br.readLine());

        for(int i=0;i<n;i++){
            st=new StringTokenizer(br.readLine());

            order=st.nextToken();

            if(order.equals("push"))  num=Integer.parseInt(st.nextToken());

            orderSelect(order, num);
        }
        System.out.print(sb);
        br.close();
    }
    static void orderSelect(String order, int num){
        switch(order){
            case "push":
            stack.push(num);
            break;

            case "top":
            if(stack.isEmpty()) sb.append(-1).append("\n");
            else sb.append(stack.peek()).append("\n");
            break;

            case "size":
            sb.append(stack.size()).append("\n");
            break;

            case "empty":
            if(stack.isEmpty()) sb.append(1).append("\n");
            else sb.append(0).append("\n");
            break;

            case "pop":
            if(stack.isEmpty()) sb.append(-1).append("\n");
            else sb.append(stack.pop()).append("\n");
            break;
        }
    }
}
